File: src/main/java/org/sample/ElasticsearchClient.java
package org.sample;

import org.elasticsearch.common.settings.Settings;
import org.elasticsearch.node.InternalSettingsPreparer;
import org.elasticsearch.node.Node;
import org.elasticsearch.node.NodeValidationException;
import org.elasticsearch.plugins.Plugin;
import org.elasticsearch.transport.Netty4Plugin;

import java.io.IOException;
import java.util.Arrays;
import java.util.Collection;

public class ElasticsearchClient {

    static Node node = null;

    public static Node startEmbeddedElasticSearch() throws NodeValidationException {
        node = new MyNode(
                Settings.builder()
                        .put("transport.type", "netty4")
                        .put("http.type", "netty4")
                        .put("http.enabled", "true")
                        .put("path.home", "elasticsearch-data")
                        .build(), Arrays.asList(Netty4Plugin.class));

        node.start();
        return node;
    }

    static class MyNode extends Node {
        public MyNode(Settings preparedSettings, Collection<Class<? extends Plugin>> classpathPlugins) {
            super(InternalSettingsPreparer.prepareEnvironment(preparedSettings, null), classpathPlugins);
        }
    }

    public static void stopEmbeddedElasticSearch() throws IOException {
        if (node != null) node.close();
    }
}

File: src/main/scala/org/sample/ElasticsearchClient.scala
package org.sample

import java.util.{Arrays, Collection}

import org.elasticsearch.common.settings.Settings
import org.elasticsearch.node.Node
import org.elasticsearch.node.InternalSettingsPreparer.{prepareEnvironment => env}
import org.elasticsearch.plugins.Plugin
import org.elasticsearch.transport.Netty4Plugin

class MyNode(settings: Settings, plugins: Collection[Class[_ <: Plugin]]) extends Node(env(settings, null), plugins) {}

object ElasticsearchClient {
  private var node: Option[MyNode] = None

  def startEmbeddedElasticSearch() = {
    val settings = Settings.builder.put("transport.type", "netty4")
      .put("http.type", "netty4")
      .put("http.enabled", "true")
      .put("path.home", "elasticsearch-data")
      .build
    node = Some(new MyNode(settings, Arrays.asList(classOf[Netty4Plugin])))
    node.get.start()
  }

  def stopEmbeddedElasticSearch(): Unit = node.foreach(_.close())
}
